base/multidimensional.jl

```diff
@@ -835,33 +832,23 @@ ensure_indexable(I::Tuple{}) = ()
 @inline to_indices(A, I::Tuple{Vararg{Union{Integer, CartesianIndex}}}) = to_indices(A, (), I)
 # But some index types require more context spanning multiple indices
 # CartesianIndex is unfolded outside the inner to_indices for better inference
-@inline function to_indices(A, inds, I::Tuple{CartesianIndex{N}, Vararg{Any}}) where N
-    _, indstail = IteratorsMD.split(inds, Val(N))
-    (map(i -> to_index(A, i), I[1].I)..., to_indices(A, indstail, tail(I))...)
-end
+_to_indices1(A, inds, I1::CartesianIndex) = map(Fix1(to_index, A), I1.I)
+_cutdim(inds, I1::CartesianIndex{N}) where {N} = IteratorsMD.split(inds, Val(N))[2]
```

Comment:
Is it reasonable to define `ndims` for `CartesianIndex` and `AbstractArray{CartesianIndex}` and use them here instead of `Val(N)`?